core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -505,12 +506,27 @@ private boolean joinElectedMaster(DiscoveryNode masterNode) {
         }
     }
 
+    private void submitRejoin(String source) {
```

Comment:
shall we use this in org.elasticsearch.discovery.zen.ZenDiscovery#publish as well?